lib/compress/huf_compress.c

```diff
@@ -1244,6 +1235,51 @@ typedef struct {
 #define SUSPECT_INCOMPRESSIBLE_SAMPLE_SIZE 4096
 #define SUSPECT_INCOMPRESSIBLE_SAMPLE_RATIO 10  /* Must be >= 2 */
 
+unsigned HUF_minTableLog(size_t srcSize, unsigned maxSymbolValue)
+{
+    U32 minBitsSrc = ZSTD_highbit32((U32)(srcSize)) + 1;
+    U32 minBitsSymbols = ZSTD_highbit32(maxSymbolValue) + 2;
+    U32 minBits = minBitsSrc < minBitsSymbols ? minBitsSrc : minBitsSymbols;
+    if (minBits < FSE_MIN_TABLELOG) minBits = FSE_MIN_TABLELOG;
+    assert(srcSize > 1); /* Not supported, RLE should be used instead */
+    return minBits;
+}
+
+unsigned HUF_optimalTableLog(unsigned maxTableLog, size_t srcSize, unsigned maxSymbolValue, void* workSpace, size_t wkspSize, HUF_CElt* table, const unsigned* count, HUF_depth_mode depthMode)
```

Comment:
Some interface implementation detail : 
`HUF_CElt* table` : when expressed this way, it _implies_ that `table` is an expected output of the function.

But it's not.
Effectively, `table` is only provided as a kind of temporary workspace, anything it may contain is just thrown away afterwards.

How to make the difference ? Well, to follow the established convention, `table` should not exist as a separate parameter, but be blended inside `workSpace`.

Now, I appreciate that it's probably easier to employ `workSpace` and `HUF_CElt*` separately because that's how they exist from the caller side, and trying to bundle them together in a single `workSpace` might end up being more messy for the caller.

So okay, implementation complexity is a criteria.

In which case, please document clearly that `table` is just a specialized "workspace", not an expected output of the function.